File: src/main/java/pro/sky/java/weatherfrontend/controller/WeatherController.java
package pro.sky.java.weatherfrontend.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import pro.sky.java.weatherfrontend.service.WeatherService;
import reactor.core.publisher.Mono;

@Controller
@RequiredArgsConstructor
public class WeatherController {

    private final WeatherService weatherService;

    @GetMapping
    public String index() {
        return "index";
    }

    @GetMapping("/get")
    public Mono<String> getWeather(@RequestParam String city, Model model) {
        return weatherService.get(city)
                .doOnSuccess(w -> model.addAttribute("result", w))
                .thenReturn("weather");
    }
}

File: src/main/kotlin/pro/sky/java/weatherfrontend/controller/WeatherController.kt
package pro.sky.java.weatherfrontend.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import pro.sky.java.weatherfrontend.html.indexPage
import pro.sky.java.weatherfrontend.html.weatherPage
import pro.sky.java.weatherfrontend.service.WeatherService

@RestController
class WeatherController(
    private val weatherService: WeatherService
) {
    @GetMapping
    fun index(): String {
        return indexPage()
    }

    @GetMapping("/get")
    suspend fun getWeather(@RequestParam city: String): String {
        return weatherPage(weatherService.get(city))
    }
}